components/automate-cli/cmd/chef-automate/summary.go

```diff
@@ -0,0 +1,457 @@
+
+// Run Status summary
+func (ss *Summary) Prepare(sshUtil SSHUtil) error {
+	automateIps, chefServerIps, opensearchIps, postgresqlIps, errList := ss.checkIPAddresses()
+	if errList != nil && errList.Len() > 0 {
+		return status.Wrap(getSingleErrorFromList(errList), status.ConfigError, ipAddressError)
+	}
+	err := ss.prepareFEScript(sshUtil, automateIps, "Automate", "FE")
+	if err != nil {
+		return err
+	}
+	err = ss.prepareFEScript(sshUtil, chefServerIps, "Chef Server", "FE")
+	if err != nil {
+		return err
+	}
+	err = ss.prepareBEScript(sshUtil, opensearchIps, "OpenSearch", "BE")
+	if err != nil {
+		return err
+	}
+	err = ss.prepareBEScript(sshUtil, postgresqlIps, "Postgresql", "BE")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// get sshConfig
+func (ss *Summary) getSSHConfig() SSHUtil {
+	sshconfig := &SSHConfig{}
+	sshconfig.sshUser = ss.infra.Outputs.SSHUser.Value
+	sshconfig.sshKeyFile = ss.infra.Outputs.SSHKeyFile.Value
+	sshconfig.sshPort = ss.infra.Outputs.SSHPort.Value
+	sshUtil := NewSSHUtil(sshconfig)
+	return sshUtil
+}
+
+func (ss *Summary) prepareBEScript(sshUtil SSHUtil, serviceIps []string, serviceName, serviceType string) error {
+	script := ""
+	err := ss.executeCommandForArrayofIPs(sshUtil, serviceIps, script, serviceName, serviceType)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (ss *Summary) prepareFEScript(sshUtil SSHUtil, serviceIps []string, serviceName, serviceType string) error {
+	script := GenerateOriginalAutomateCLICommand(
+		&cobra.Command{
+			Use: "chef-automate",
+		}, []string{
+			"status",
+			"-t",
+			"150",
+		})
+
+	err := ss.executeCommandForArrayofIPs(sshUtil, serviceIps, script, serviceName, serviceType)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (ss *Summary) executeCommandForArrayofIPs(sshUtil SSHUtil, remoteIps []string, script, serviceName, serviceType string) error {
+	getConfigJsonString := convTfvarToJson(a2haHabitatAutoTfvars)
+	for i := 0; i < len(remoteIps); i++ {
+		sshUtil.getSSHConfig().hostIP = remoteIps[i]
+		if serviceType == "FE" {
+			status := ss.getFEStatus(sshUtil, script, remoteIps[i], serviceName)
+			ss.feStatus = append(ss.feStatus, status)
+		}
+		if serviceType == "BE" {
+			authToken, err := ss.readA2haHabitatAutoTfvarsAuthToken(getConfigJsonString)
+			if err != nil {
+				return err
+			}
+			status := ss.getBEStatus(sshUtil, remoteIps[i], authToken, serviceName)
+			ss.beStatus = append(ss.beStatus, status)
+		}
+
+	}
+	return nil
+}
+
+func (ss *Summary) readA2haHabitatAutoTfvarsAuthToken(getConfigJsonString string) (string, error) {
+	config := A2haHabitatAutoTfvars{}
+	err := json.Unmarshal([]byte(getConfigJsonString), &config)
+	if err != nil {
+		return "", err
+	}
+	authToken := config.HabSupHttpGatewayAuthToken
+
+	return authToken, nil
+}
+
+func (ss *Summary) getBEStatus(sshUtil SSHUtil, ip string, authToken, serviceType string) BeStatusValue {
+	var service, role, health, formatted, serviceState, servicePid string
```

Comment:
This function is doing a lot can we break it into functions with meaningful name

Reply:
Done